plugins/defaultplugins/plugin_impl_vpp.go

```diff
@@ -471,83 +450,28 @@ func (plugin *Plugin) initIF(ctx context.Context) error {
-		Log:          bfdLogger,
-		GoVppmux:     plugin.GoVppmux,
-		ServiceLabel: plugin.ServiceLabel,
-		SwIfIndexes:  plugin.swIfIndexes,
-		BfdIDSeq:     1,
-		Stopwatch:    stopwatch,
+	// BFD configurator
+	plugin.bfdConfigurator = &ifplugin.BFDConfigurator{}
+	bfdLogger := plugin.Log.NewLogger("-bfd-conf")
+	if err := plugin.bfdConfigurator.Init(plugin.PluginName, bfdLogger, plugin.GoVppmux, plugin.swIfIndexes, plugin.enableStopwatch); err != nil {
+		return err
 	}
-	plugin.bfdConfigurator.Init(plugin.bfdSessionIndexes, plugin.bfdAuthKeysIndexes, plugin.bfdEchoFunctionIndex, BfdRemovedAuthKeys)
-
 	plugin.Log.Debug("bfdConfigurator Initialized")
 
-	if plugin.enableStopwatch {
-		stopwatch = measure.NewStopwatch("stnConfigurator", stnLogger)
-	}
-
-	plugin.stnAllIndexes = nametoidx.NewNameToIdx(stnLogger, plugin.PluginName, "stn-all-indexes", nil)
-	plugin.stnUnstoredIndexes = nametoidx.NewNameToIdx(stnLogger, plugin.PluginName, "stn-unstored-indexes", nil)
-
-	plugin.stnConfigurator = &ifplugin.StnConfigurator{
-		Log:                 bfdLogger,
-		GoVppmux:            plugin.GoVppmux,
-		SwIfIndexes:         plugin.swIfIndexes,
-		StnUnstoredIndexes:  plugin.stnUnstoredIndexes,
-		StnAllIndexes:       plugin.stnAllIndexes,
-		StnUnstoredIndexSeq: 1,
-		StnAllIndexSeq:      1,
-		Stopwatch:           stopwatch,
-	}
-	if err := plugin.stnConfigurator.Init(); err != nil {
+	// STN configurator
+	plugin.stnConfigurator = &ifplugin.StnConfigurator{}
+	stnLogger := plugin.Log.NewLogger("-stn-conf")
+	if err := plugin.stnConfigurator.Init(plugin.PluginName, stnLogger, plugin.GoVppmux, plugin.swIfIndexes, plugin.enableStopwatch); err != nil {
 		return err
 	}
-
 	plugin.Log.Debug("stnConfigurator Initialized")
 
 	// NAT indices
-	plugin.sNatIndices = nametoidx.NewNameToIdx(natLogger, plugin.PluginName, "snat-indices", nil)
-	plugin.sNatMappingIndices = nametoidx.NewNameToIdx(natLogger, plugin.PluginName, "snat-mapping-indices", nil)
-	plugin.dNatIndices = nametoidx.NewNameToIdx(natLogger, plugin.PluginName, "dnat-indices", nil)
-	plugin.dNatStMappingIndices = nametoidx.NewNameToIdx(natLogger, plugin.PluginName, "dnat-st-mapping-indices", nil)
-	plugin.dNatIdMappingIndices = nametoidx.NewNameToIdx(natLogger, plugin.PluginName, "dnat-id-mapping-indices", nil)
-
-	plugin.natConfigurator = &ifplugin.NatConfigurator{
-		Log:                  natLogger,
-		GoVppmux:             plugin.GoVppmux,
-		SwIfIndexes:          plugin.swIfIndexes,
-		SNatIndices:          plugin.sNatIndices,
-		SNatMappingIndices:   plugin.sNatMappingIndices,
-		DNatIndices:          plugin.dNatIndices,
-		DNatStMappingIndices: plugin.dNatStMappingIndices,
-		DNatIdMappingIndices: plugin.dNatIdMappingIndices,
-		NatIndexSeq:          1,
-		NatMappingTagSeq:     1,
-		Stopwatch:            stopwatch,
-	}
-	if err := plugin.natConfigurator.Init(); err != nil {
+	plugin.natConfigurator = &ifplugin.NatConfigurator{}
+	natLogger := plugin.Log.NewLogger("-nat-conf")
```

Comment:
It is now not really necessary to keep this here. We can initialize loggers inside `Init` functions, since they are only really used inside its package.